### PR TITLE
edge: pull out RTCPeerConnection shim

### PR DESCRIPTION
**Pull request by fippo**
Consider this an intent-to-ship. This needs some more QA and a proper package version on npm

pulls out the RTCPeerConnection shim into a separate package.

Moving forward this will make it easier to publish new patch versions
of that package without depending on an adapter patch release.

Also this makes it a lot easier to set up project infrastructure
like code coverage separately.

Coverage with unit tests has gone from 60% to >90%

### CODE REVIEW

**Comment by fippo** [comment]
@KaptenJansson as discussed. Not ready for merge yet

File: package.json [hunk 1]
```diff
@@ -19,7 +19,8 @@
     "test": "grunt && mocha test/unit && node test/run-tests.js"
   },
   "dependencies": {
-    "sdp": "^1.5.0"
+    "sdp": "^1.5.0",
+    "rtcpeerconnection-shim": "git://github.com/fippo/rtcpeerconnection-shim.git#0.0.1"
```
Comment (KaptenJansson) [inline package.json]: No NPM?

Comment (fippo) [reply package.json]: will publish to npm once i'm sure that i will no longer nuke the repo :-)

**Comment by fippo** [comment]
amazing. All but one test (icegatheringstatechange) still pass. And that one just seems a bit flaky.

**Comment by agouaillard-cosmo** [comment]
I am concerned about code provided under CLA moving our to external private repository. Can we do the same arch changes while keeping the code under google's control?

**Comment by ggarber** [comment]
LGTM 

As one of the (small) contributors to that code I'm totally ok with moving that under @fippo's control as he is the main author of that component and showed the commitment to maintain it and accept contributions.

Otherwise fippo can create a fork and everybody will use his fork anyway, so better to not keep an obsolete version in this repository IMO and keep core adapter as simple as possible.

**Comment by agouaillard** [comment]
I'm concerned about the commitment to accept contributions, or to see its contributions reviewed of those who blocks other contributors GitHub account. I think a reviewer has a great power, which comes with a great responsibility to treat everybody equally, to accept and review all contributions, and to address disagreement or issues, when any arise, through communication.

**Comment by fippo** [comment]
@aboba kindly agreed to act as a reviewer should this become an issue.

**Comment by agouaillard** [comment]
unblocked for less than a week, and then blocked again, without a single thread of communication. I think my point is publicly made.

**Comment by fippo** [comment]
This PR is hardly the place to discuss why I blocked you on twitter or on github (as disclosed by you previously).

It creates a restriction for contributing as @agouaillard but you have @agouaillard-cosmo's that still has ability to open PRs and issues on any repositories that I own as well as getting github to send me notification emails.

**Comment by agouaillard** [comment]
I don't see why you blocking me would put the burden on me to change my GitHub handle with which I have been contributing and reviewing many repositories for ages.

**Comment by agouaillard** [comment]
I don't see why you blocking me would put the burden on me to change my GitHub handle with which I have been contributing and reviewing many repositories for ages.
I m treating everybody with equal respect and professionalism whatever the account I use. I can only regret you choose not too. It is on your tracking record, not mine. I reached out to the owners of this repository to express my opinion. They will decide what they want to do with their code.

**Comment by KaptenJansson** [comment]
LGTM and I approve for the 0.0001% code that I've contributed to this particular shim :).

If in the event some issue arise around contributing to the shim, we will work it out in a professional manner. It's in everyone's interest in keeping it up to date and functional.

**Pull request merged**